### DISCUSSION TITLE
fsm x-ref $refs buttons not working

### DISCUSSION THREAD

**Discussion by markfirmware** [alpinejs/alpine#950] [1. Help] [answered]
https://codepen.io/markfirmware/pen/QWKqORX

I expect the buttonsto change the state variable but it does not.

**Comment by KevinBatdorf** [comment, marked as answer]
You should update it to `@click=“state = 1”`. `$refs.identifier` doesn’t give you access to the component properties, but the actual element itself (to which you can then access whatever you need on that element).

**Comment by markfirmware** [comment]
Thanks, I will use x-id/$component instead of x-ref/$refs.
